src/ngraph/pass/pass.hpp

```diff
@@ -45,6 +46,17 @@ namespace ngraph
             FOP_FUSIONS = 0x4,
             ALL_FUSIONS = 0xFFFFFFFF
         };
+        enum class PassProperty : uint32_t
+        {
+            //`DIFFERENTIABLE_FUSIONS` produce ops that support autodiff
```

Comment:
Let's leave things out until they are actually being used by something.
I am a little worried that using the enum is micro-optimizing and that sets would work just as well here.

Reply:
If you are referring to `DIFFERENTIABLE_FUSIONS`, etc. sure I can remove them for now till we change the passes and merge with the `FusionType` above. 

In terms of enums, their usage here is not really for performance or optimization. I think for this use case (a set of true/false flags), it does fit the bill quite well. For anything more complex, we should use something different. Sure we could use a `set` or some other container for them. But bit mask works really well with enums. For example, if we want to look up a combination of flags (A|B|C), look up using a `set` is not very natural. Using enum is really just trying to keep things simple.

Reply:
I was referring to the unused ones; we can fill them in as we know what we actually need. I suspect we won't need those.

Reply:
Fixed.